flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/table/catalog/hive/util/HiveTableUtil.java

```diff
@@ -96,11 +106,104 @@ public class HiveTableUtil {
 
     private HiveTableUtil() {}
 
-    public static TableSchema createTableSchema(
+    /** Create a Flink's Schema by hive client. */
+    public static org.apache.flink.table.api.Schema createSchema(
+            HiveConf hiveConf,
+            Table hiveTable,
+            HiveMetastoreClientWrapper client,
+            HiveShim hiveShim) {
+
+        Tuple4<List<FieldSchema>, List<FieldSchema>, Set<String>, Optional<UniqueConstraint>>
+                hiveTableInfo = extractHiveTableInfo(hiveConf, hiveTable, client, hiveShim);
+
+        return createSchema(
+                hiveTableInfo.f0,
+                hiveTableInfo.f1,
+                hiveTableInfo.f2,
+                hiveTableInfo.f3.orElse(null));
+    }
+
+    /** Create a Flink's Schema from Hive table's columns and partition keys. */
+    public static org.apache.flink.table.api.Schema createSchema(
+            List<FieldSchema> nonPartCols,
+            List<FieldSchema> partitionKeys,
+            Set<String> notNullColumns,
+            @Nullable UniqueConstraint primaryKey) {
+        return Schema.newBuilder()
+                .fromResolvedSchema(
+                        createResolvedSchema(
```

Comment:
In here, we will fisrt convert to `ResolvedSchema` and then convert to `Schema`. So why not convert to `Schema` directly?
I push a [commmit](https://github.com/luoyuxia/flink/commit/52c5f234099b7268f182c130baa8b9bb183e3bd0) ,  PTAL. 
It decides on you to take or not.

Reply:
> In here, we will fisrt convert to ResolvedSchema and then convert to Schema. So why not convert to Schema directly?

It's intend to save the code, because creating `ResolvedSchema` and `Schema` actually doing the same thing. IMO, we do not have to avoid using `Schema.newBuilder().fromResolvedSchema().build()` as much as possible, I think its harmless. Since it is a bridge to covert the ResolvedSchema back to Schema when needed.

In this case, we can create `ResolvedSchema` directly from the Hive table information. In my original thought, we even do not have to add the dedicated method as below, the caller just can do it depend on the requirements.

```
Schema createScheam(List<FieldSchema> nonPartCols,
            List<FieldSchema> partitionKeys,
            Set<String> notNullColumns,
            @Nullable UniqueConstraint primaryKey) {
      return Schema.newBuilder()
                .fromResolvedSchema(
                        createResolvedSchema(
                                nonPartCols, partitionKeys, notNullColumns, primaryKey))
                .build();
}
```

Reply:
I aggree it's not harmful to do extra schema conversion althogth it'll create extra object and `createSchema` will be  called a liitle of frequently.
But if we can meet both code save and reduce schema conversion with minor efforts, why not do that?

Reply:
Make sense, will apply your commits :)